code/controllers/subsystem/ticker.dm

```diff
@@ -703,17 +708,32 @@ SUBSYSTEM_DEF(ticker)
 			//Break chain since this has a sleep input in it
 			addtimer(CALLBACK(player, TYPE_PROC_REF(/mob/dead/new_player/authenticated, make_me_an_observer)), 1)
 
-/datum/controller/subsystem/ticker/proc/SetRoundEndSound(the_sound)
+/datum/controller/subsystem/ticker/proc/SetRoundEndSound(the_sound, duration)
 	set waitfor = FALSE
 	round_end_sound_sent = FALSE
 	round_end_sound = fcopy_rsc(the_sound)
+	round_end_sound_duration = duration
+	if(round_end_sound_timer)//replace any existing timer with a new one
+		var/time_left = reboot_delay - (world.time - start_wait)
+		if(time_left > round_end_sound_duration)
+			deltimer(round_end_sound_timer)
+			round_end_sound_timer = addtimer(CALLBACK(src, PROC_REF(PlayRoundEndSound)), time_left - round_end_sound_duration, TIMER_STOPPABLE)
+		else //not enough time, play it anyway
+			deltimer(round_end_sound_timer)
+			round_end_sound_timer = null
+			PlayRoundEndSound()
```

Comment:
Still it doesn't look right to me. You're trying to play endround sound even if just when you are attempting to set a round.

Reply:
The last line? I chose to have admin sounds play regardless of whether they have enough time to fully do so, I can block them for playing if you want. Regular roundend sounds won't play if they don't have time though.

Reply:
but why in set sound proc

Reply:
Because that gets called every time an admin presses the related button as opposed to the code in reboot() which only gets called on reboot, this is to allow a new sound to be set after reboot already started and a default sound already exists